etna/pipeline/autoregressive_pipeline.py

```diff
@@ -130,8 +136,23 @@ def _forecast(self) -> TSDataset:
                     message="You probably set wrong freq.",
                     action="ignore",
                 )
-                current_ts_forecast = current_ts.make_future(current_step)
-            current_ts_future = self.model.forecast(current_ts_forecast)
+
+                if isinstance(self.model, get_args(ContextRequiredModelType)):
+                    self.model = cast(ContextRequiredModelType, self.model)
+                    if isinstance(self.model, DeepBaseModel):
```

Comment:
Nested ifs are not readable. Better to use plain if cases.

Reply:
Fixed